x/mongo/driver/topology/polling_srv_records_test.go

```diff
@@ -256,25 +249,11 @@ func TestPollSRVRecords(t *testing.T) {
 func TestPollingSRVRecordsLoadBalanced(t *testing.T) {
 	createLBTopology := func(t *testing.T, uri string) *Topology {
 		t.Helper()
-
-		cs, err := connstring.ParseAndValidate(uri)
-		assert.Nil(t, err, "connstring.ParseAndValidate error: %v", err)
-		cs.LoadBalancedSet = true
-		cs.LoadBalanced = true
-
-		topo, err := New(
-			WithConnString(func(connstring.ConnString) connstring.ConnString { return cs }),
-			WithURI(func(string) string { return cs.Original }),
-		)
+		topo, err := NewPrototype(options.Client().ApplyURI(uri).SetLoadBalanced(true))
 		assert.Nil(t, err, "topology.New error: %v", err)
 		return topo
 	}
 
-	t.Run("pollingRequired is set to false", func(t *testing.T) {
```

Comment:
I think we should remove this check. The given DNS results in multiple hosts, and the client options will [error on load balancing over multiple hosts](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/options/clientoptions.go#L202). The client options validation doesn't currently exist with `topology.WithConnString`. I'm working on the assumption that the client options layer is the more correct implementation, deprecating this test.

If we want to keep this test, we can mock it in the style of the succeeding test:

```
TestPollingSRVRecordsLoadBalanced/new_records_are_not_detected
```